app/stac_api/models.py

```diff
@@ -274,8 +274,16 @@ class Collection(
 ):
 
     class Meta:
-        indexes = [models.Index(fields=['name'], name='collection_name_idx')]
+        indexes = [
+            models.Index(fields=['name'], name='collection_name_idx'),
+            models.Index(fields=['published'], name='collection_published_idx')
+        ]
 
+    published = models.BooleanField(
+        default=True,
+        help_text=
+        "When not published the collection doesn't appear on the api/stac/v0.9/collections endpoint"
```

Comment:
Probably add in the help_text, that it is still accessible via the exact pass though, even when publishes=False, just to be clear about that.